File: JavaScript/LeetCode75/Q1448-CountGoodNodesInBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let count = 0;

    function traverse(root, max) {
        if(!root) return null;
        if(root.val >= max) {
            max = root.val;
            count++;
        }
        traverse(root.left, max);
        traverse(root.right, max);
    }

    traverse(root, root.val);
    return count;
};

const goodNodes = (root) => {
    let count = 0;

    function traverse(root, max) {
        if(!root) return null;
        if(root.val >= max) count++;

        traverse(root.left, Math.max(max, root.val));
        traverse(root.right, Math.max(max, root.val));       
    }
    traverse(root, root.val);
    return count;
}

File: Python/Leetcode75/Medium/BinaryTreeDFS/Q1448-CountGoodNodesInBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if not root:
            return 0

        self.cnt = 0
        self.dfs(root, float('-inf'))
        return self.cnt

    def dfs(self, node, max_val):
        if not node:
            return

        if node.val >= max_val:
            self.cnt += 1

        max_val = max(max_val, node.val)
        self.dfs(node.left, max_val)
        self.dfs(node.right, max_val)
